Libraries/LibJS/Runtime/RegExpObject.cpp

```diff
@@ -96,19 +96,10 @@ ErrorOr<String, ParseRegexPatternError> parse_regex_pattern(StringView pattern,
     auto utf16_pattern = Utf16String::from_utf8(pattern);
     StringBuilder builder;
 
-    // If the Unicode flag is set, append each code point to the pattern. Otherwise, append each
-    // code unit. But unlike the spec, multi-byte code units must be escaped for LibRegex to parse.
+    // AD-HOC: We need to escape multi-byte code units for LibRegex to parse.
     auto previous_code_unit_was_backslash = false;
-    for (size_t i = 0; i < utf16_pattern.length_in_code_units();) {
-        if (unicode || unicode_sets) {
-            auto code_point = code_point_at(utf16_pattern, i);
-            builder.append_code_point(code_point.code_point);
-            i += code_point.code_unit_count;
-            continue;
-        }
-
+    for (size_t i = 0; i < utf16_pattern.length_in_code_units(); ++i) {
```

Comment:
Just a question - but I had a thought from the commkt message - if the issue is LibRegex not being able to handle it, I wonder if this should instead be in LibRegex. We do have callers which do need to directly invoke LibRegex and expect the same result as what the ECMA implementation should

Reply:
I agree - I could change the `// AD-HOC` to `// FIXME` for now to better indicate there's still an issue here? I haven't really been able to pinpoint the issue in LibRegex though.

Reply:
iirc the parser in LibRegex uses GenericLexer, which can't handle unicode (or couldn't?)
Not too sure, but that's what I remember being the main cause.

Reply:
There was a PR working on a unicode GenericLexer. Can't remember what state that's in.